### DISCUSSION TITLE
Build time for Android is extremely high - any tips?

### DISCUSSION THREAD

**Discussion by smartyw** [organicmaps/organicmaps#595] [Q&A]
Performing a clean build of the app for Android on my Macbook takes over 30 minutes. I was wondering how the main team work and whether anyone would be prepared to write a short guide for developers that are new to the project, with tips on the best way to work including what tools are best and how they are configured?

**Comment by biodranik** [comment]
It's hard to build faster for clean builds. You may build only one android arch by specifying `-Parm64` param to the gradle.
Also repeated builds will be faster if you install ccache and configure it for 10Gb cache.

**Comment by smartyw** [comment]
Excellent tips, thank you. I've not worked with the NDK before. It might be time for me to buy a faster computer :-)

**Comment by smartyw** [comment]
Do the core team use Android Studio as an IDE? I find it takes forever "Updating symbols" and starts to run out of memory. I'm now using Visual Studio Code... but missing Android Studio. Any tips about getting Android Studio to work in an acceptable way with the app would also be appreciated.

**Comment by biodranik** [comment]
There are some memory settings/tunings in Android Studio, you can google them, probably on stack overflow. And then test and share that worked :)

**Comment by smartyw** [comment]
Maybe I'll just use vi ;-)

**Comment by rtsisyk** [comment]
IntelliJ/Android Studuo is just crap in 2021.
Unfortunately, there are no alternatives if you need a debugger on Android.
JetBrains has weird segmentation of their products. For example, IntelliJ
supports Java debugging, but doesn't support C++ debugging. CLion supports
C++ debugging, but doesn't support Java debugging. Only Android Studio can
do both.

Unfortunately, Google is still not able to create M1 build. JetBrains has
IntelliJ for M1, but not Android Studio. That means that Android Studio is
slow as a hell on the fastest laptop available on the market.

The only option is to use VScode or even vim :-)

**Comment by smartyw** [comment]
LOL. Yes, that does seem to be the case. In fact, I use VS Code for pretty much everything else anyway so I shouldn't care so much about Android. All will be well :-)

**Reply by rtsisyk** [reply]
I think that this question is answered. @smartyw, could you please mark this question as answered? Please let us know if you still need help.
